include/scippp/solving_statistics.hpp

```diff
@@ -0,0 +1,180 @@
+#pragma once
+
+#include "scippp/statistics.hpp"
+
+#include <scip/scip_solvingstats.h>
+
+// the content of this file is auto-generated by extract_solvingstats.py
+
+namespace scippp::statistics {
+
+//! number of branch and bound runs performed, including the current run
+static const Statistic<int> N_RUNS { &SCIPgetNRuns };
```

Comment:
Isn't the C++ compiler able to automatically deduce the template argument here and below?

Reply:
Probably yes, what would be the benefit?

Reply:
At least Sonar argues as follows

> Since C++17, class template arguments can be automatically deduced by the compiler, either by looking at the arguments of the class constructors or by using an explicitly defined deduction guide.
>
> Using the class template argument deduction allows to:
>
>   - Avoid verbose specification of all template parameter types for a class template.
>    - Avoid writing helper function that only serves the purpose of deducing the type of a class from its arguments. For example, std::make_pair.
>    - Be able to instantiate a class template with hard to spell or unutterable names, such as the closure type of a lambda.
>
> This rule raises an issue when explicit class template arguments that can be automatically deduced are specified.

Reply:
Does not compile

```
include/scippp/solving_statistics.hpp:12:14: error: too few template arguments for class template 'Statistic'
static const Statistic<> N_RUNS { &SCIPgetNRuns };
```